app/models/service_template_ansible_tower.rb

```diff
@@ -6,6 +6,22 @@ class ServiceTemplateAnsibleTower < ServiceTemplate
   alias_method :job_template, :configuration_script
   alias_method :job_template=, :configuration_script=
 
+  def self.create_catalog_item(options, _auth_user = nil)
+    transaction do
+      create(options.except(:config_info)) do |service_template|
+        config_info = validate_config_info(options)
+
+        service_template.job_template = if config_info[:configuration_script_id]
```

Comment:
@mkanoor this is the refactoring work for existing AnsibleTower. The work you're interested is https://github.com/ManageIQ/manageiq/pull/13600